include/components/MovableCamera.hxx

```diff
@@ -38,24 +41,21 @@ class MovableCamera : public Urho3D::Camera {
     float move_speed = DEFAULT_MOVE_SPEED;
     float mouse_sensitivity = DEFAULT_MOVE_SENSITIVITY;
 
-    std::unordered_map<int, Urho3D::Vector3> key_map;
+    std::array<std::pair<int, Urho3D::Vector3>, 4> key_map{{{DEFAULT_FORWARD, Urho3D::Vector3::FORWARD},
+                                                            {DEFAULT_BACK, Urho3D::Vector3::BACK},
+                                                            {DEFAULT_LEFT, Urho3D::Vector3::LEFT},
+                                                            {DEFAULT_RIGHT, Urho3D::Vector3::RIGHT}}};
 
     float m_yaw = 0.f;
     float m_pitch = 0.f;
 
-    int forward_key = DEFAULT_FORWARD;
-    int back_key = DEFAULT_BACK;
-    int left_key = DEFAULT_LEFT;
-    int right_key = DEFAULT_RIGHT;
-
   public:
-    enum specialKey { FORWARD, BACK, LEFT, RIGHT };
+    enum struct MoveKey { FORWARD, BACK, LEFT, RIGHT };
 
     explicit MovableCamera(Urho3D::Context* context);
     static void RegisterObject(Urho3D::Context* context);
     void move(float delta_time);
-    void setKey(int key, const Urho3D::Vector3& vec) { key_map[key] = vec; }
-    void unsetKey(int key) { key_map.erase(key); }
-    void setSpecialKey(specialKey special_key, int new_key);
+    void setKey(MoveKey key, int new_key) { key_map.at(static_cast<int>(key)).first = new_key; }
+    void unsetKey(MoveKey key) { key_map.at(static_cast<int>(key)).first = -1; }
```

Comment:
Did you purposefully use exception-throwing methods or is it an oversight?

Reply:
Clang tidy asked me to do so :)

Reply:
ClangTidy is just a tool to help you think about possible errors; it does not always know what you really want